concent_api/core/subtask_helpers.py

```diff
@@ -168,20 +168,21 @@ def are_keys_and_addresses_unique_in_message_subtask_results_accepted(subtask_re
     is_provider_ethereum_address_unique = (len(unique_provider_ethereum_addresses) == 1)
     is_provider_ethereum_public_key_unique = (len(unique_provider_ethereum_public_keys) == 1)
 
-    requestor_public_key = subtask_results_accepted_list[0].task_to_compute.requestor_public_key
-    are_all_signed_by_requestor = all(
-        is_golem_message_signed_with_key(
-            hex_to_bytes_convert(requestor_public_key),
-            subtask_results_accepted
-        ) for subtask_results_accepted in subtask_results_accepted_list
-    )
-
     return (
-        are_all_signed_by_requestor and
         is_requestor_public_key_unique and
         is_requestor_ethereum_address_unique and
         is_requestor_ethereum_public_key_unique and
         is_provider_public_key_unique and
         is_provider_ethereum_address_unique and
         is_provider_ethereum_public_key_unique
     )
+
+def are_subtask_results_accepted_messages_sign_by_the_same_requestor(subtask_results_accepted_list: List[SubtaskResultsAccepted]) -> bool:
+    requestor_public_key = subtask_results_accepted_list[0].task_to_compute.requestor_public_key
```

Comment:
Shouldn't you check if list has at least one element ?

Reply:
We have that check elsewhere. Look at the `test_provider_send_force_payment_with_empty_subtask_results_accepted_list_concent_should_return_http_400` in `core/tests/test_integration_force_payment.py` it returns 
`b'{"error": "subtask_results_accepted_list must be a list type and contains at least one message", "error_code": "message.value_wrong_length"}'`